node/src/app/cli.rs

```diff
@@ -228,7 +228,15 @@ impl Cli {
                     ReactorExit::ProcessShouldContinue => info!("finished joining"),
                 }
 
-                let config = joiner_runner.into_inner().into_validator_config().await?;
+                let (joiner_reactor, joiner_queue) = joiner_runner.into_inners();
+                let config = joiner_reactor.into_validator_config().await?;
+
+                // At this point, the joiner is shut down, so we clear the queue to ensure any
+                // connections whose handshake completed but have not been registered are dropped.
```

Comment:
Isn't dropping the event queue achieving the same effect?

Reply:
No, we only keep handles to it unfortunately.